_rules/element-lang-matches-common-language-off6ek.md

```diff
@@ -0,0 +1,323 @@
+---
+id: off6ek
+name: HTML element language subtag matches language
+rule_type: atomic
+description: |
+  This rule checks that the primary language subtag of an element matches its language
+accessibility_requirements:
+  wcag20:3.1.2: # Language of Parts (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+  wcag-technique:H58: # Using language attributes to identify changes in the human language
+    forConformance: false
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: satisfied
+input_aspects:
+  - DOM tree
+  - Accessibility tree
+  - CSS Styling
+  - Language
+acknowledgments:
+  authors:
+    - Jean-Yves Moyen
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # using aria-labelledby instead
+  - 'alt-require'
+---
+
+## Applicability
+
+This rule applies to any HTML element for which all the following are true:
+
+- the element is an [inclusive descendant][] in the [flat tree][] of a `body` element; and
+- The element is in a [document][] with a [content type][] of `text/html`; and
+- The element has a `lang` [attribute value][] which is a [valid language tag][]; and
+- The element has some non-empty [text with the same programmatic language][].
+
+## Expectation
+
+For each test target, the [primary language][] of its `lang` [attribute value][] is a [common language][] of the test target.
+
+## Assumptions
+
+- This rule assumes that user agents and assistive technologies can programmatically determine [valid language tags][] even if these do not conform to the [BCP 47][] syntax.
+
+- This rule assumes that [grandfathered tags][] are not used as these will not be recognized as [valid language tags][].
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- Related rule:
+  - [_Element with `lang` Attribute Has Valid Language Tag_](https://act-rules.github.io/rules/de46e4)
+- [Understanding Success Criterion 3.1.2: Language of Page][usc312]
+- [H58: Using language attributes to identify changes in the human language](https://www.w3.org/WAI/WCAG21/Techniques/html/H58)
+- [BCP 47: Tags for Identifying Languages](https://www.ietf.org/rfc/bcp/bcp47.txt)
+- [The `lang` and `xml:lang` attributes](https://html.spec.whatwg.org/multipage/dom.html#the-lang-and-xml:lang-attributes)
+
+In all examples, the `html` element has itself a `lang` attribute in order to make sure that the examples do not fail [Success Criterion 3.1.1 Language of Page](https://www.w3.org/TR/WCAG21/#language-of-page). These `html` elements are, however, never applicable because they are not descendants of a `body` element, and the example descriptions do not mention them further.
```

Comment:
```suggestion
In all examples, the `html` element has itself a `lang` attribute in order to make sure that the examples satisfy [Success Criterion 3.1.1 Language of Page](https://www.w3.org/TR/WCAG21/#language-of-page). These `html` elements are, however, never applicable because they are not descendants of a `body` element, and the example descriptions do not mention them further.
```